test/WebJobs.Script.Tests/Diagnostics/LinuxAppServiceEventGeneratorTests.cs

```diff
@@ -32,7 +35,12 @@ public LinuxAppServiceEventGeneratorTests()
             var loggerFactoryMock = new Mock<LinuxAppServiceFileLoggerFactory>(MockBehavior.Strict);
             loggerFactoryMock.Setup(f => f.GetOrCreate(It.IsAny<string>())).Returns<string>(s => _loggers[s]);
 
-            _generator = new LinuxAppServiceEventGenerator(loggerFactoryMock.Object);
+            var environmentMock = new Mock<IEnvironment>();
+            environmentMock.Setup(f => f.GetEnvironmentVariable(It.Is<string>(v => v == "WEBSITE_HOSTNAME")))
+                .Returns<string>(s => _hostNameDefault);
+
+            var hostNameProvider = new HostNameProvider(environmentMock.Object);
+            _generator = new LinuxAppServiceEventGenerator(loggerFactoryMock.Object, hostNameProvider);
```

Comment:
This is a huge nit -- but rather than using Mocks for IEnvironment, you can use TestEnvironment which allows you to build up the dictionary of values and pass it in. It's a little easier to read when going through these tests. No real need to switch this, but FYI for next time.

Reply:
Ooh. Ok, I need to do a final test today anyway. I will update that too if i get a chance.